rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -69,21 +69,43 @@
 
 namespace
 {
-
 //==============================================================================
-// A function to take ros topic names and convert them to valid Zenoh keys.
+// A function that generates a key expression for message transport of the format
+// <ros_domain_id>/<topic_name>/<topic_type>/<topic_hash>
 // In particular, Zenoh keys cannot start or end with a /, so this function
 // will strip them out.
 // The Zenoh key is also prefixed with the ros_domain_id.
 // Performance note: at present, this function allocates a new string and copies
 // the old string into it. If this becomes a performance problem, we could consider
 // modifying the topic_name in place. But this means we need to be much more
 // careful about who owns the string.
-z_owned_keyexpr_t ros_topic_name_to_zenoh_key(const char * const topic_name, size_t domain_id)
-{
-  const std::string keyexpr_str = std::to_string(domain_id) +
-    "/" +
-    rmw_zenoh_cpp::liveliness::mangle_name(topic_name);
```

Comment:
Just to make sure I understand properly; we are no longer using `mangle_name` here so that we stick with slashes wherever possible for performance reasons, right?  The downside here is that it won't be possible to look at a keyexpression and tell which parts are the topic name, and which are Zenoh splits, but I guess I'm OK with that for now.

Reply:
Yes that's exactly right.